sdv/cag/one_hot_encoding.py

```diff
@@ -88,6 +90,14 @@ def _fit(self, data, metadata):
         """
         pass
 
+    def _get_updated_metadata(self, metadata):
+        table_name = self._get_single_table_name(metadata)
+        for column in self._column_names:
+            if metadata.tables[table_name].columns[column]['sdtype'] == 'categorical':
+                metadata.tables[table_name].columns[column]['sdtype'] = 'numerical'
```

Comment:
Instead of modifying the input metadata directly, we should either deepcopy it before doing this assignment or convert it to a dict, update the metadata, then return a new metadata object.